internal/app/machined/pkg/runtime/v1alpha1/bootloader/sdboot/sdboot.go

```diff
@@ -235,6 +250,13 @@ func (c *Config) KexecLoad(r runtime.Runtime, disk string) error {
 			}
 		}
 
+		efiCtx := efivario.NewDefaultContext()
+
+		// clear the selected entry
+		if err := WriteVariable(efiCtx, LoaderEntrySelectedName, ""); err != nil {
```

Comment:
why do we clear selected entry in KexecLoad? should we clear this on bootloader install?

Reply:
ah good point

Reply:
that doesn't seem to work when booting off iso, since sd-boot already set that and we can't clear it, so doing at kexec makes sense, since we only need to be cleared for kexec

Reply:
```text
updating EFI variables
[    3.975603] Error: failed to install bootloader: efivario/set: write /sys/firmware/efi/efivars/LoaderEntrySelected-4A67B082-0A4C-41CF-B6C7-440B29BB8C4F: invalid argument

```

Reply:
hmmm weird this only fails when booting off iso